### PR TITLE
Implements sorting by array content.

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**

### CODE REVIEW

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java [hunk 1]
```diff
@@ -353,4 +355,12 @@ public static void deleteCollection(Firestore database, String collectionToDelet
       throw new FirebaseException(e.toString());
     }
   }
+
+  public static String queryByArrayField(Firestore database, String collectionName, String arrayName,
```
Comment (MikolJiron) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java:369]: Add javadoc.

Comment (MikolJiron) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java:369]: Add javadoc.

**Review by MikolJiron — CHANGES_REQUESTED** [review]

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
LGTM. Don't forget to rebase devel.

**Review by MikolJiron — APPROVED** [review]

**Review by CarlosM10011 — APPROVED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java [hunk 2]
```diff
@@ -353,4 +354,23 @@ public static void deleteCollection(Firestore database, String collectionToDelet
...
+   * @param collectionName
+   * @param arrayName
+   * @param arrayFieldValue
+   * @return All documents in database collection specified by collectionName that containes
+   * arrayFieldValue in their arrayName field, which is supposed to be an array.
+   * @throws FirebaseException
+   * @throws NoSuchElementException
+   */
+  public static String queryByArrayField(Firestore database, String collectionName, String arrayName,  
+      Optional<String> arrayFieldValue) throws FirebaseException, NoSuchElementException {
```
Comment (Lorraine-Lyu) [inline src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java:370]: So if I throw all possible errors in this way, I don't need to catch them, right?

Comment (tinagrasso) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java:370]: Right, but somewhere up the call chain, you'll need to handle them.

**Review by tinagrasso — APPROVED** [review]

File: src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java (hunk 2 above)
Comment (Lorraine-Lyu) [reply src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java:370]: Yeah I handled them in endpoint servlets/

**Pull request merged**